src/openrct2/windows/options.c

```diff
@@ -1707,16 +1709,18 @@ static void window_options_invalidate(rct_window *w)
         break;
 
     case WINDOW_OPTIONS_PAGE_MISC:
-        // unknown park flag can disable real name checkbox
-        if (gParkFlags & PARK_FLAGS_LOCK_REAL_NAMES_OPTION)
+        // The real name setting of clients is fixed to that of the server
+        // and the server cannot change the setting during gameplay to prevent desyncs
+        if (network_get_mode() != NETWORK_MODE_NONE) {
```

Comment:
If we're going to disable settings in runtime, there should be at least an explanation tooltip why a setting is disabled